### PR TITLE
[browser] Fix fingerprinting of relinked dotnet.js - follow up

### PR DESCRIPTION
**Pull request by maraf**
Address comments from https://github.com/dotnet/runtime/pull/86048
- More relaxed regex to check files are fingerprinted
- Test turning on fingerprinting
- Use `ResolvedRuntimePack` to get runtime pack version

### CODE REVIEW

File: src/mono/nuget/Microsoft.NET.Sdk.WebAssembly.Pack/build/Microsoft.NET.Sdk.WebAssembly.Browser.targets [hunk 1]
```diff
@@ -190,14 +190,13 @@ Copyright (c) .NET Foundation. All rights reserved.
...
       <_WasmConfigFileCandidates Include="@(StaticWebAsset)" Condition="'%(SourceType)' == 'Discovered'" />
-      <_DotNetJsItem Include="@(ReferenceCopyLocalPaths)" Condition="('%(FileName)' == 'dotnet' or '%(FileName)' == 'dotnet.native') and '%(Extension)' == '.js'" />
 
       <!-- Remove dotnet.js/wasm from runtime pack, in favor of the relinked ones in @(WasmNativeAsset) -->
       <ReferenceCopyLocalPaths Remove="@(ReferenceCopyLocalPaths)"
                                Condition="@(WasmNativeAsset->Count()) > 0 and ( '%(FileName)' == 'dotnet' or '%(FileName)' == 'dotnet.native' ) and ('%(Extension)' == '.wasm' or '%(Extension)' == '.js')" />
     </ItemGroup>
     <PropertyGroup>
-      <_DotNetJsBuildVersion>%(_DotNetJsItem.NuGetPackageVersion)</_DotNetJsBuildVersion>
+      <_DotNetJsBuildVersion>%(ResolvedRuntimePack.NuGetPackageVersion)</_DotNetJsBuildVersion>
```
Comment (radical) [inline src/mono/nuget/Microsoft.NET.Sdk.WebAssembly.Pack/build/Microsoft.NET.Sdk.WebAssembly.Browser.targets:199]: `_DotNetJsBuildVersion` -> `_RuntimePackVersion`

File: src/mono/wasm/Wasm.Build.Tests/Blazor/BuildPublishTests.cs [hunk 2]
```diff
@@ -65,15 +65,21 @@ public void DefaultTemplate_NoAOT_WithWorkload(string config)
...
 
     [Theory]
-    [InlineData("Debug")]
-    [InlineData("Release")]
-    public void DefaultTemplate_BuildNative_WithWorkload(string config)
+    [InlineData("Debug", false)]
+    [InlineData("Release", false)]
+    [InlineData("Debug", true)]
+    [InlineData("Release", true)]
+    public void DefaultTemplate_BuildNative_WithWorkload(string config, bool expectFingerprintOnDotnetJs)
```
Comment (radical) [inline src/mono/wasm/Wasm.Build.Tests/Blazor/BuildPublishTests.cs:72]:
```suggestion
    public void DefaultTemplate_Build_CheckFingerprinting(string config, bool expectFingerprintOnDotnetJs)
```

Comment (radical) [reply src/mono/wasm/Wasm.Build.Tests/Blazor/BuildPublishTests.cs:72]: We need the same thing for *publish* too, right?

File: src/mono/wasm/Wasm.Build.Tests/Blazor/BuildPublishTests.cs [hunk 3]
```diff
@@ -65,15 +66,26 @@ public void DefaultTemplate_NoAOT_WithWorkload(string config)
...
     {
         string id = $"blz_buildandbuildnative_{config}_{Path.GetRandomFileName()}";
 
         CreateBlazorWasmTemplateProject(id);
 
-        BlazorBuild(new BlazorBuildOptions(id, config, NativeFilesType.Relinked), "/p:WasmBuildNative=true");
+        void Execute(Func<BlazorBuildOptions, string[], (CommandResult, string)> target) => target(
+            new BlazorBuildOptions(id, config, NativeFilesType.Relinked, ExpectFingerprintOnDotnetJs: expectFingerprintOnDotnetJs),
+            new[] {
+                "/p:WasmBuildNative=true",
```
Comment (radical) [inline src/mono/wasm/Wasm.Build.Tests/Blazor/BuildPublishTests.cs:82]: This is needed only for build.

**Review by radical — APPROVED** [review]

File: src/mono/nuget/Microsoft.NET.Sdk.WebAssembly.Pack/build/Microsoft.NET.Sdk.WebAssembly.Browser.targets [hunk 4]
```diff
@@ -187,18 +187,20 @@ Copyright (c) .NET Foundation. All rights reserved.
 
   <Target Name="ResolveWasmOutputs" DependsOnTargets="_ResolveWasmOutputs" />
 
-  <Target Name="_ResolveWasmOutputs" DependsOnTargets="ResolveReferences;PrepareResourceNames;ComputeIntermediateSatelliteAssemblies;_ResolveWasmConfiguration;_WasmNativeForBuild">
+  <Target Name="_GetWasmRuntimePackVersion">
+    <PropertyGroup>
+      <_WasmRuntimePackVersion>%(ResolvedRuntimePack.NuGetPackageVersion)</_WasmRuntimePackVersion>
+    </PropertyGroup>
```
Comment (radical) [inline src/mono/nuget/Microsoft.NET.Sdk.WebAssembly.Pack/build/Microsoft.NET.Sdk.WebAssembly.Browser.targets:193]: Also, error here if we couldn't find a version.

File: src/mono/wasm/Wasm.Build.Tests/Blazor/BuildPublishTests.cs (hunk 3 above)
Comment (maraf) [reply src/mono/wasm/Wasm.Build.Tests/Blazor/BuildPublishTests.cs]: Sorry, misread the error message. It check whether after publish the `for-build` is empty

Comment (radical) [reply src/mono/wasm/Wasm.Build.Tests/Blazor/BuildPublishTests.cs]: Ah, that would be the case for Debug. Ok, ignore my comment then.

**Pull request merged**